release-lock-file-spec.md

```diff
@@ -90,117 +224,374 @@ Sources are declared in a key/value mapping.
-    * Type: String
-    * Format: Hex encoded ethereum address of the deployed contract.
-* `deploy_transaction`:
-    * Required: No
-    * Type: String
-    * Format: [BIP122](https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki) URI which defines the transaction in which this contract was created.
-* `deploy_block`:
-    * Required: No
-    * Type: String
-    * Format: [BIP122](https://github.com/bitcoin/bips/blob/master/bip-0122.mediawiki) URI which defines the block in which this contract was created.
-* `link_dependencies`:
-    * Reqired: Required to have an entry for each link reference found in either the `bytecode` or `runtime_bytecode` fields.
-    * Type: Object
-    * Format:
-        * All keys **must** be strings which are formatted as valid link targets.
-        * All values **must** conform to *one of* the following formats:
-            * A hex encoded ethereum address.
-            * A [json pointer](https://tools.ietf.org/html/rfc6901) to another *Deployment* object in the release lock file.  The `address` from the resolved *Deployment* object should be used.
-            * An IPFS URI with a JSON point in the fragment portion of the URI.  The IPFS hash must resolves to a valid release lock file.  The json pointer **must** resolves to a *Deployment* object within the release lock file.  The `address` from the resolved *Deployment* object should be used.
-
-
-#### BIP122 URIs
+    * Keys **must** be valid package names matching the regular expression `[a-z][-a-z0-9]{0,213}`
+    * Values **must** be valid IPFS URIs which resolve to a valid *Release Lock File*
+
+
+## Object Definitions
+
+Definitions for different objects used within the release lockfile.  All
+objects allow custom fields to be included.  Custom fields **should** be
+prefixed with `x-` to prevent name collisions with future versions of the
+specification.
+
+
+### The *Package Meta* Object
+
+The *Package Meta* object is defined to have the following key/value pairs.
+
+
+#### Authors: `authors`
+
+The `authors` field defines a list of human readable names for the authors of
+this package.  Release lockfiles **may** include this field. 
+
+* Required: No
+* Key: `authors`
+* Type: List of Strings
+
+
+### License: `license`
+
+The `license` field declares the license under which this package is released.
+This value **should** be conform to the
+[SPDX](https://en.wikipedia.org/wiki/Software_Package_Data_Exchange) format.
+Release lockfiles **should** include this field.
+
+* Required: No
+* Key: `license`
+* Type: String
+
+
+### Description: `description`
+
+The `description` field provides additional detail that may be relevant for the
+package.  Release lockfiles **may** include this field.
+
+* Required: No
+* Key: `description`
+* Type: String
+
+
+### Keywords: `keywords`
+
+The `keywords` field provides relevant keywords related to this package.  
+
+* Required: No
+* Key: `keywords`
+* Type: List of Strings
+
+
+### Links: `links`
```

Comment:
but isn't the meta object referenced from the lockfile?
why not referencing the lockfile from the meta file and solve the meta file on a higher level where we introduce nondeterminism?